api/model/src/main/java/org/projectnessie/model/types/ContentTypeRegistry.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model.types;
+
+import org.projectnessie.model.Content;
+
+/** An implementation of this interface is passed to {@link ContentTypeBundle}s. */
+public interface ContentTypeRegistry {
+  void register(Class<? extends Content> type);
```

Comment:
The Json part is fine, I think. My concern is only about the Open API spec (which has its own discriminators in the schema annotation on `Content`).

Since we now expect custom types to be present, is it worth declaring a sub-set of possible types as `oneOf` in our Open API spec?

Reply:
Define "possible types" ;)

The problem is that we cannot dynamically extend that `oneOf` + the OpenAPI discriminators. OTOH, it's only "an issue" for new/completely-custom-private types. New "core" types will naturally be added there.

Reply:
Why would we want to keep `oneOf` in the Open API spec at all? With pluggable types, the spec is guaranteed to be skewed sooner or later. 

I'm proposing to replace it with a generic `Object` type (or the Open API) equivalent.

Reply:
I think this could become a problem for the future Event API. The generated JSON schema for this API currently has this:

```json
      "Content" : {
        "title" : "Content",
        "type" : "object",
        "properties" : {
          "id" : {
            "type" : "string"
          }
        },
        "oneOf" : [ {
          "$ref" : "#/components/schemas/IcebergTable"
        }, {
          "$ref" : "#/components/schemas/DeltaLakeTable"
        }, {
          "$ref" : "#/components/schemas/IcebergView"
        }, {
          "$ref" : "#/components/schemas/Namespace"
        } ],
        "discriminator" : {
          "propertyName" : "type",
          "mapping" : {
            "ICEBERG_TABLE" : "#/components/schemas/IcebergTable",
            "DELTA_LAKE_TABLE" : "#/components/schemas/DeltaLakeTable",
            "ICEBERG_VIEW" : "#/components/schemas/IcebergView",
            "NAMESPACE" : "#/components/schemas/Namespace"
          }
        }
      },
```

The `oneOf` approach will be problematic if someone uses this schema to attempt a polymorphic deserialization of components of different types, for example when using `KafkaJsonSchemaDeserializer` to deserialize events from a Kafka topic.

Even if the consumer has the jar containing the custom content implementation, the schema won't allow it, and deserializing such a content is likely to fail imho.

(However, I _think_ this could work when using a schema-less deserializer like `KafkaJsonDeserializer`.)

Reply:
Hm - custom content-types aren't new though. Not sure why events infra needs to know the whole model.

Reply:
> perhaps replace with anyOf(coreType1, coreType2, ..., Object).

`anyOf` seems to work